### PR TITLE
Updated project dev dependencies

### PR DESCRIPTION
**Pull request by emravera**
Updated changed project dev dependencies. Also updated the test framework used and updated supported nodejs versions.

### CODE REVIEW

File: deps-injector/deps-injector.js
Comment (nestornav) [inline deps-injector/deps-injector.js]: We have the simple quotes style. Please check.

**Comment by nestornav** [comment]
Great work here! That huge upgrade will help a lot to improve the project. Seems there are a conflict onto your branch, please check it.

File: deps-injector/deps-injector.js
Comment (emravera) [reply deps-injector/deps-injector.js]: Thanks!

**Review by Undre4m — APPROVED** [review]
Looks great, a much needed update! Should be ready for merge after resolving conflicts.

**Review by nestornav — APPROVED** [review]

**Comment by Undre4m** [comment]
Should be up to date with **master**. Lots of merging conflicts so second review would be advised before merging.

**Comment by emravera** [comment]
I think we can merge this. It was updated with master, so should be no conflicts.

**Comment by emravera** [comment]
I am working on the merge once I update I let you know. Discard the other message.

**Comment by Undre4m** [comment]
I rebased for you. All conflicts should be resolved. A second review is needed JIC but should be ready for merge at current state.

**Comment by emravera** [comment]
Now looks great! Thanks @Undre4m

**Pull request merged**
